docs/man/man3/Corpus.3
.TH "Corpus" 3 "Fri May 14 2021" "Version 2.1.0" "Text-Classifier" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Corpus \- This class describes a corpus\&.  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include <corpus\&.hpp>\fP
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBCorpus\fP (void)"
.br
.RI "Constructs a new instance\&. "
.ti -1c
.RI "\fBCorpus\fP (std::string name, std::string inputFile)"
.br
.RI "Constructs a new instance\&. "
.ti -1c
.RI "\fB~Corpus\fP (void)"
.br
.RI "Destroys the object\&. "
.ti -1c
.RI "std::string \fBget_Name\fP (void) const"
.br
.RI "Gets the name\&. "
.ti -1c
.RI "std::string \fBget_OutputFileName\fP (void) const"
.br
.RI "Gets the output file name\&. "
.ti -1c
.RI "std::string \fBget_InputFile\fP (void) const"
.br
.RI "Gets the input file\&. "
.ti -1c
.RI "void \fBset_Name\fP (std::string newName)"
.br
.RI "Sets the name\&. "
.ti -1c
.RI "void \fBset_OutputFileName\fP (std::string newOutputFileName)"
.br
.RI "Sets the output file name\&. "
.ti -1c
.RI "void \fBset_InputFile\fP (std::string newInputFile)"
.br
.RI "Sets the input file\&. "
.ti -1c
.RI "void \fBgenerateCorpus\fP (std::vector< std::string > &stopWords, \fBPreProcesser\fP &preProcesser)"
.br
.RI "Generates and stores the corpus data\&. "
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "std::string \fBname_\fP"
.br
.ti -1c
.RI "std::string \fBoutputFileName_\fP"
.br
.ti -1c
.RI "std::string \fBinputFile_\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
This class describes a corpus\&. 
.PP
Definition at line 40 of file corpus\&.hpp\&.
.SH "Constructor & Destructor Documentation"
.PP 
.SS "Corpus::Corpus (std::string name, std::string inputFile)"

.PP
Constructs a new instance\&. 
.PP
\fBParameters\fP
.RS 4
\fIname\fP The name 
.br
\fIinputFile\fP The input file 
.RE
.PP

.PP
Definition at line 43 of file corpus\&.cpp\&.
.PP
.nf
43                                                    {
44     set_Name(name);
45     set_OutputFileName("\&.\&./outputs/corpus_" + name + "\&.txt");
46     set_InputFile(inputFile);
47 }
.fi
.SH "Member Function Documentation"
.PP 
.SS "void Corpus::generateCorpus (std::vector< std::string > & stopWords, \fBPreProcesser\fP & preProcesser)"

.PP
Generates and stores the corpus data\&. 
.PP
\fBParameters\fP
.RS 4
\fIstopWords\fP The stop words 
.br
\fIpreProcesser\fP The pre processer 
.RE
.PP

.PP
Definition at line 115 of file corpus\&.cpp\&.
.PP
.nf
115                                                                                         {
116     Chrono myChrono;
117     int dataLines;
118     myChrono\&.startChrono();
119     std::string outputFile = "\&.\&./outputs/preProcesserHelper\&.txt";
120     std::string dataType = get_Name() + ",";
121     dataLines = preProcesser\&.loadData(inputFile_, dataType);
122     preProcesser\&.convertLowerCase();
123     preProcesser\&.erasePunctuationSigns();
124     preProcesser\&.eraseAllNumbers();
125     preProcesser\&.storeData(outputFile, 0);
126     preProcesser\&.eraseReservedWords(stopWords, outputFile);
127     preProcesser\&.storeData(outputFileName_, dataLines);
128     myChrono\&.stopChrono();
129     std::cout << std::endl << "Elapsed time for corpus " << name_ << ": " << myChrono\&.get_Seconds(5) << " seconds\&." << std::endl;
130 }
.fi
.SS "std::string Corpus::get_InputFile (void) const"

.PP
Gets the input file\&. 
.PP
\fBReturns\fP
.RS 4
The input file\&. 
.RE
.PP

.PP
Definition at line 78 of file corpus\&.cpp\&.
.PP
.nf
78                                            {
79     return inputFile_;
80 }
.fi
.SS "std::string Corpus::get_Name (void) const"

.PP
Gets the name\&. 
.PP
\fBReturns\fP
.RS 4
The name\&. 
.RE
.PP

.PP
Definition at line 60 of file corpus\&.cpp\&.
.PP
.nf
60                                       {
61     return name_;
62 }
.fi
.SS "std::string Corpus::get_OutputFileName (void) const"

.PP
Gets the output file name\&. 
.PP
\fBReturns\fP
.RS 4
The output file name\&. 
.RE
.PP

.PP
Definition at line 69 of file corpus\&.cpp\&.
.PP
.nf
69                                                 {
70     return outputFileName_;
71 }
.fi
.SS "void Corpus::set_InputFile (std::string newInputFile)"

.PP
Sets the input file\&. 
.PP
\fBParameters\fP
.RS 4
\fInewInputFile\fP The new input file 
.RE
.PP

.PP
Definition at line 105 of file corpus\&.cpp\&.
.PP
.nf
105                                                   {
106     inputFile_ = newInputFile;
107 }
.fi
.SS "void Corpus::set_Name (std::string newName)"

.PP
Sets the name\&. 
.PP
\fBParameters\fP
.RS 4
\fInewName\fP The new name 
.RE
.PP

.PP
Definition at line 87 of file corpus\&.cpp\&.
.PP
.nf
87                                         {
88     name_ = newName;
89 }
.fi
.SS "void Corpus::set_OutputFileName (std::string newOutputFileName)"

.PP
Sets the output file name\&. 
.PP
\fBParameters\fP
.RS 4
\fInewOutputFileName\fP The new output file name 
.RE
.PP

.PP
Definition at line 96 of file corpus\&.cpp\&.
.PP
.nf
96                                                             {
97     outputFileName_ = newOutputFileName;
98 }
.fi


.SH "Author"
.PP 
Generated automatically by Doxygen for Text-Classifier from the source code\&.


src/corpus.cpp
/*=======================================================================================
=========================================================================================
	=                                                                              =
	=            Proyect:       Text Classifier                                    =
	=            File name:     corpus.cpp                                         =
	=            Author:        Adrián Epifanio Rodríguez Hernández                =
	=            Date:          21/04/2021                                         =
	=            Subject:       Advanced Artificial Inteligence                    =
	=            Language:      C++                                                =
	=            Email:         alu0101158280@ull.edu.es                           =
	=            Place:         Universidad De La Laguna                           =
	=                           Escuela Superior de Ingeniería y Tecnología        =
	=                                                                              =
=========================================================================================
=======================================================================================*/
/*
* @Author: Adrian Epifanio
* @Date:   2021-04-28 12:35:30
* @Last Modified by:   Adrian Epifanio
* @Last Modified time: 2021-04-30 12:56:35
*/
/*------------------  FUNCTIONS  -----------------*/

#include "../include/corpus.hpp"

/*------------------------------------------------*/

/**
 * @brief      Constructs a new instance.
 */
Corpus::Corpus (void) {
	set_Name("");
	set_OutputFileName("");
	set_InputFile("");
}

/**
 * @brief      Constructs a new instance.
 *
 * @param[in]  name       The name
 * @param[in]  inputFile  The input file
 */
Corpus::Corpus (std::string name, std::string inputFile) {
	set_Name(name);
	set_OutputFileName("../outputs/corpus_" + name + ".txt");
	set_InputFile(inputFile);
}

/**
 * @brief      Destroys the object.
 */
Corpus::~Corpus (void) {
}

/**
 * @brief      Gets the name.
 *
 * @return     The name.
 */
std::string Corpus::get_Name (void) const {
	return name_;
}

/**
 * @brief      Gets the output file name.
 *
 * @return     The output file name.
 */
std::string Corpus::get_OutputFileName (void) const {
	return outputFileName_;
}

/**
 * @brief      Gets the input file.
 *
 * @return     The input file.
 */
std::string Corpus::get_InputFile (void) const {
	return inputFile_;
}

/**
 * @brief      Sets the name.
 *
 * @param[in]  newName  The new name
 */
void Corpus::set_Name (std::string newName) {
	name_ = newName;
}

/**
 * @brief      Sets the output file name.
 *
 * @param[in]  newOutputFileName  The new output file name
 */
void Corpus::set_OutputFileName (std::string newOutputFileName) {
	outputFileName_ = newOutputFileName;
}

/**
 * @brief      Sets the input file.
 *
 * @param[in]  newInputFile  The new input file
 */
void Corpus::set_InputFile (std::string newInputFile) {
	inputFile_ = newInputFile;
}

/**
 * @brief      Generates and stores the corpus data.
 *
 * @param      stopWords     The stop words
 * @param[in]  preProcesser  The pre processer
 */
void Corpus::generateCorpus (std::vector<std::string>& stopWords, PreProcesser& preProcesser) {
	Chrono myChrono;
	myChrono.startChrono();
	std::string outputFile = "../outputs/preProcesserHelper.txt";
	std::string dataType = get_Name() + ",";
	preProcesser.loadData(inputFile_, dataType);
	preProcesser.convertLowerCase();
	preProcesser.erasePunctuationSigns();
	preProcesser.eraseAllNumbers();
	preProcesser.storeData(outputFile);
	preProcesser.eraseReservedWords(stopWords, outputFile);
	preProcesser.storeData(outputFileName_);
	myChrono.stopChrono();
	std::cout << std::endl << "Elapsed time for corpus " << name_ << ": " << myChrono.get_Seconds(5) << " seconds." << std::endl;
}


